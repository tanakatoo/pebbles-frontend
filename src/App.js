import './styles/app.css';
import NavBar from "./components/common/NavBar"
import "./styles/app.css"
import { useSelector } from 'react-redux';
import MyRoutes from "./MyRoutes"

function App() {
  //get the font from the store
  const font = useSelector(state => state.langFont.font)

  //we have to hardcode the string of the font because tailwind doesn't know what to use at build
  return (
    <div className={`App ${font == 'font-poppins' ? 'font-poppins' : 'font-notoJP'}`}>
      <NavBar />
      <MyRoutes />
    </div >
  );
}

export default App;
