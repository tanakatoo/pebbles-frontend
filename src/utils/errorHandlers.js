import { rest } from 'msw'

let loginResponse = {
    token: 'abcdefg',
    user: {
        id: 2,
        username: 'hello',
        role: 'regular',
        premium_join_date: '',
        premium_end_date: '',
        free_trial_start_date: '',
    }
}



export const errorHandlers = [

    //login
    rest.post('http://localhost:3001/auth/login', (req, res, ctx) => {
        const failLogin = {
            "error": {
                "message": "INVALID_CREDENTIALS",
                "status": 401
            }
        }
        console.log(res(ctx.json([failLogin])))
        return res(ctx.json([failLogin]));
    }),

    rest.post('http://localhost:3001/email', (req, res, ctx) => {
        console.log('MOCK error /email');
        const noData = {
            "error": {
                "message": "no data",
                "status": 400
            }
        }
        return res(ctx.json(noData));
    }),

]