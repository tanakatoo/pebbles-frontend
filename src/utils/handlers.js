import { rest } from 'msw'

let loginResponse = {
    token: 'abcdefg',
    user: {
        id: 2,
        username: 'hello',
        role: 'regular',
        premium_join_date: '',
        premium_end_date: '',
        free_trial_start_date: '',
    }
}

const contactsResponse =
    [
        {
            "user_id": 1,
            "username": "ktoo",
            "avatar": "2.jpg"
        },
        {
            "user_id": 15,
            "username": "june",
            "avatar": "4.jpg"
        }
    ]

const blockedResponse = [
    {
        "blocked_user_id": 3,
        "username": "blockMe",
        "avatar": "9.jpg"
    }
]
const mainSearchResults = [
    {
        "username": "hello2",
        "avatar": "9.jpg",
        "name": null,
        "about": null,
        "premium_acct_id": null,
        "end_date": null
    },
    {
        "username": "helloo",
        "avatar": "14.jpg",
        "name": null,
        "about": null,
        "premium_acct_id": null,
        "end_date": null
    }
]

const changePassword = "completed"

const newMessage = [
    {
        "toavatar": "9.jpg",
        "to": "hello",
        "fromavatar": "",
        "from": "hello"
    }
]

const continuingMsg = [
    {
        "toavatar": "9.jpg",
        "fromavatar": "2.jpg",
        "from": "ktoo",
        "to": "hello",
        "from_user_id": 1,
        "to_user_id": 2,
        "msg": "the most recent one reply to second one",
        "sent_at": "2023-06-09 15:30:00",
        "read": true
    },
    {
        "toavatar": "2.jpg",
        "fromavatar": "9.jpg",
        "from": "hello",
        "to": "ktoo",
        "from_user_id": 2,
        "to_user_id": 1,
        "msg": "2nd most recent one reply",
        "sent_at": "2023-06-08 15:30:00",
        "read": true
    },
    {
        "toavatar": "9.jpg",
        "fromavatar": "2.jpg",
        "from": "ktoo",
        "to": "hello",
        "from_user_id": 1,
        "to_user_id": 2,
        "msg": "first message",
        "sent_at": "2023-06-07 15:30:00",
        "read": true
    }
]

const latestMessages = [
    {
        "toavatar": "9.jpg",
        "fromavatar": "4.jpg",
        "from": "june",
        "to": "hello",
        "from_user_id": 15,
        "to_user_id": 2,
        "msg": "testing\n",
        "sent_at": "2023-07-15 20:57:44.804634",
        "read": false
    },
    {
        "toavatar": "9.jpg",
        "fromavatar": "2.jpg",
        "from": "ktoo",
        "to": "hello",
        "from_user_id": 1,
        "to_user_id": 2,
        "msg": "the most recent one reply to second one",
        "sent_at": "2023-06-09 15:30:00",
        "read": true
    }
]

const returnUser = [
    {
        "id": 2,
        "username": "hello",
        "name": null,
        "avatar": "9.jpg",
        "country_en": null,
        "country_ja": null,
        "city_en": null,
        "city_ja": null,
        "state_en": null,
        "state_ja": null,
        "gender": "female",
        "about": null,
        "free_trial_start_date": null,
        "study_buddy_bio": "just chatting",
        "study_buddy_purpose": "i don't know why",
        "native_language": "Japanese",
        "learning_language": "English",
        "language_level": "Beginner",
        "time_zone": "NewYork",
        "age_range": "26-35",
        "study_buddy_active": true,
        "study_buddy_activate_date": "2023-07-13T04:00:00.000Z"
    }
]

export const handlers = [

    //login
    rest.post('http://localhost:3001/auth/login', (req, res, ctx) => {

        console.log(res(ctx.json(loginResponse)))
        return res(ctx.json(loginResponse));
    }),

    rest.post('http://localhost:3001/auth/logout', (req, res, ctx) => {

        console.log('MOCK /logout')
        window.localStorage.setItem('token', null)
        return res(ctx.json());
    }),


    // Called via <ExploreCommunity type="studyBuddy" />
    rest.get('http://localhost:3001/study-buddies/1', (req, res, ctx) => {
        console.log('MOCK /study-buddies/1');

        return res(ctx.json([]));
    }),


    rest.post('http://localhost:3001/auth/register', (req, res, ctx) => {
        console.log('MOCK /auth/register');

        return res(ctx.json(loginResponse));
    }),


    rest.post('http://localhost:3001/email', (req, res, ctx) => {
        console.log('MOCK /email');

        return res(ctx.json('sents'));
    }),


    rest.get('http://localhost:3001/users/contacts', (req, res, ctx) => {
        console.log('MOCK /users/contacts');

        return res(ctx.json(contactsResponse));
    }),

    rest.get('http://localhost:3001/users/blocked', (req, res, ctx) => {
        console.log('MOCK /users/blocked');

        return res(ctx.json(blockedResponse));
    }),

    rest.get('http://localhost:3001/users/search', (req, res, ctx) => {
        console.log('MOCK /users/search');

        return res(ctx.json(mainSearchResults));
    }),

    rest.post('http://localhost:3001/auth/change-password', (req, res, ctx) => {
        console.log('MOCK /auth/change-password');

        return res(ctx.json(changePassword));
    }),

    //for continuing conversation
    rest.get('http://localhost:3001/messages/:username', (req, res, ctx) => {
        console.log('MOCK /messages/:username');
        console.log(req.params)
        return res(ctx.json(continuingMsg));
    }),

    //get latest message from all users
    rest.get('http://localhost:3001/messages', (req, res, ctx) => {
        console.log('MOCK /messages');
        return res(ctx.json(latestMessages));
    }),

    //get latest message from all users
    rest.get('http://localhost:3001/users/saved-users', (req, res, ctx) => {
        console.log('MOCK /users/saved-users');
        return res(ctx.json(returnUser));
    }),
]