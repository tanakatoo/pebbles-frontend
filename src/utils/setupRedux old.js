
import { configureStore } from '@reduxjs/toolkit';
import rootReducer from "../reducers/rootReducer"
import { persistStore, persistReducer, FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER } from 'redux-persist';

import storage from 'redux-persist/lib/storage'
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'

export const persistConfig = {
    key: 'root', // Key prefix for the persisted state
    storage, // Storage mechanism (e.g., local storage)
    stateReconciler: autoMergeLevel2, // Specify which parts of the state to persist
    whitelist: ['profile', 'langFont']
};

export const persistedReducer = persistReducer(persistConfig, rootReducer);
export const store = configureStore({
    reducer: persistedReducer,
    middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
            serializableCheck: {
                ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER]
            }
        })
})
export const persistedStore = persistStore(store)