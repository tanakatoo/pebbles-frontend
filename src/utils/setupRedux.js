import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
// As a basic setup, import your same slice reducers
import { persistStore, persistReducer, FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER } from 'redux-persist';
import { PersistGate } from 'redux-persist/integration/react'
import { ListSlice } from "../reducers/rootReducer";
import rootReducer from "../reducers/rootReducer"
import storage from 'redux-persist/lib/storage'
import storeSetup from "./storeSetup"
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'

// const persistConfig = {
//     key: 'root', // Key prefix for the persisted state
//     storage, // Storage mechanism (e.g., local storage)
//     stateReconciler: autoMergeLevel2, // Specify which parts of the state to persist
//     whitelist: ['profile', 'langFont']
// };
// const persistedReducer = persistReducer(persistConfig, rootReducer);

// const persistedStore = persistStore(store)
// const root = ReactDOM.createRoot(document.getElementById('root'));


export function renderWithProviders(
    ui,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        store = storeSetup(preloadedState),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {

        return <Provider store={store}>{children}</Provider>;
    }

    // Return an object with the store and all of RTL's query functions
    return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}