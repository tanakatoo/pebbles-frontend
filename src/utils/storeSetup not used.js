import { combineReducers, configureStore } from '@reduxjs/toolkit'

import rootReducer from '../reducers/rootReducer'
// import userReducer from '../features/users/userSlice'

// Create the root reducer separately so we can extract the RootState type
// const rootReducer = combineReducers({
//     user: userReducer
// })

export const setupStore = preloadedState => {
    return configureStore({
        reducer: rootReducer,
        preloadedState
    })
}