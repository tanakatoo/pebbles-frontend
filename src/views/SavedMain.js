import React from 'react'
import MainCard from '../components/saved/MainCard'
import usePageText from '../hooks/usePageText'
import PageTitle from '../components/common/PageTitle'
import { useSelector } from 'react-redux'

function SavedMain() {
    const pageText = usePageText('saved')
    const lang = useSelector(state => state.langFont.lang)
    return (
        <div className='px-2 border-t border-t-gray-stroke'>
            <div className='flex flex-col justify-center mb-12'>
                <PageTitle text={pageText.TITLE} extraClasses='my-3' />
                <div className='shrink-0 flex flex-wrap gap-8 mx-8 justify-center
            '>

                    <div className='flex justify-center'>
                        <MainCard img='bg-usersSquare'
                            text={pageText.USERS}
                            link={'users'}
                        />
                    </div>
                    <div className='flex justify-center'>
                        <MainCard img='bg-markeplaceSquare'
                            text={pageText.MARKETPLACE}
                        />
                    </div>
                    <div className='flex justify-center'>
                        <MainCard img='bg-regionalSquare'
                            text={pageText.REGIONAL_RESOURCES}
                        />
                    </div>
                    <div className='flex justify-center'>
                        <MainCard img='bg-regionalParisSquare'
                            text={pageText.REGIONAL_GROUP}
                        />
                    </div>
                    <div className='flex justify-center'>
                        <MainCard img='bg-learningSquare'
                            text={pageText.TOWN_RESOURCES}
                        />
                    </div>
                    <div className='flex justify-center'>
                        <MainCard img='bg-learningGlobeSquare'
                            text={pageText.TOWN_GROUPS}
                        />
                    </div>

                </div>
            </div>
        </div>
    )
}

export default SavedMain