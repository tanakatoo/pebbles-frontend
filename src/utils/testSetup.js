// import { persistedStore, store } from './setupRedux';
import { render } from '@testing-library/react'
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import { setupTestStore } from './setupRedux';


const renderWithProviders = (ui, options = {}) => {
    const [store, persistedStore] = setupTestStore(options.preloadedState);

    const AllProviders = ({ children }) => {
        return (
            <Provider store={store}>
                <PersistGate loading={null} persistor={persistedStore}>
                    {children}
                </PersistGate>
            </Provider>
        )
    };

    return { store, ...render(ui, { wrapper: AllProviders, ...options }) };
}


// "re-export everything"
export * from '@testing-library/react';

export { renderWithProviders };