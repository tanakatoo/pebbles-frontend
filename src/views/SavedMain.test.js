import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { fireEvent, renderWithProviders } from '../utils/testSetup';
import SavedMain from './SavedMain';
import App from '../App';
import { waitFor } from '../utils/testSetup';
import { act } from 'react-dom/test-utils';


// Mock window.scrollTo()
window.scrollTo = jest.fn();
afterAll(() => {
    jest.clearAllMocks();
});

beforeEach(async () => {
    const { getByTestId, getByLabelText, getByText } = renderWithProviders(
        <MemoryRouter >
            <App />
        </MemoryRouter>
    );

    // Populate form
    let changeLang = getByText('EN', { exact: true });
    fireEvent.click(changeLang);
})

afterEach(() => {
    window.localStorage.token = null
})

describe('SavedMain', () => {
    test('renders without crashing', () => {
        renderWithProviders(
            <MemoryRouter>
                <SavedMain />
            </MemoryRouter>
        );
    });

    test('matches snapshot', async () => {
        const { asFragment } = renderWithProviders(
            <MemoryRouter>
                <SavedMain />
            </MemoryRouter>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    test('Click on users and displays users that were saved', async () => {
        const { getByTestId, getByLabelText, getByText } = await renderWithProviders(
            <MemoryRouter >
                <SavedMain />
            </MemoryRouter>
        );


        // Submit form
        // await act(async () => {
        //     fireEvent.click(await getByText(/Users/));
        // })
        fireEvent.click(await getByText(/Users/));
        // fireEvent.click(getByTestId(/loginFormLogin/));

        // Wait for pageText to be populated
        await waitFor(() => {
            expect(getByTestId('pageTitle')).not.toBeEmptyDOMElement();
        });

    });

});