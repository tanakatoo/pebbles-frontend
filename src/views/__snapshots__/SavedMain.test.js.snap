// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedMain matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="px-2 border-t border-t-gray-stroke"
  >
    <div
      class="flex flex-col justify-center mb-12"
    >
      <h1
        class="my-3 py-4 px-2 text-center text-mobile-header-2 font-bold text-primary-dark"
        data-testid="pageTitle"
      >
        Saved items
      </h1>
      <div
        class="shrink-0 flex flex-wrap gap-8 mx-8 justify-center "
      >
        <div
          class="flex justify-center"
        >
          <a
            href="/users"
          >
            <div
              class="drop-shadow-md shrink-0 relative w-[200px] h-[200px] rounded-ml bg-usersSquare bg-cover flex flex-col items-center justify-center"
            >
              <div
                class="bg-white/50 absolute rounded-ml w-full h-full"
              />
              <span
                class="drop-shadow-xl text-center z-10 text-primary-dark text-mobile-page-header font-medium bg-white/75 px-2 rounded-sm w-full py-2"
              >
                Users
              </span>
            </div>
          </a>
        </div>
        <div
          class="flex justify-center"
        >
          <a
            href="/"
          >
            <div
              class="drop-shadow-md shrink-0 relative w-[200px] h-[200px] rounded-ml bg-markeplaceSquare opacity-50 cursor-default bg-cover flex flex-col items-center justify-center"
            >
              <div
                class="bg-white/50 absolute rounded-ml w-full h-full"
              />
              <span
                class="drop-shadow-xl text-center z-10 text-primary-dark text-mobile-page-header font-medium bg-white/75 px-2 rounded-sm w-full py-2"
              >
                Marketplace items
              </span>
            </div>
          </a>
        </div>
        <div
          class="flex justify-center"
        >
          <a
            href="/"
          >
            <div
              class="drop-shadow-md shrink-0 relative w-[200px] h-[200px] rounded-ml bg-regionalSquare opacity-50 cursor-default bg-cover flex flex-col items-center justify-center"
            >
              <div
                class="bg-white/50 absolute rounded-ml w-full h-full"
              />
              <span
                class="drop-shadow-xl text-center z-10 text-primary-dark text-mobile-page-header font-medium bg-white/75 px-2 rounded-sm w-full py-2"
              >
                Regional info center resources
              </span>
            </div>
          </a>
        </div>
        <div
          class="flex justify-center"
        >
          <a
            href="/"
          >
            <div
              class="drop-shadow-md shrink-0 relative w-[200px] h-[200px] rounded-ml bg-regionalParisSquare opacity-50 cursor-default bg-cover flex flex-col items-center justify-center"
            >
              <div
                class="bg-white/50 absolute rounded-ml w-full h-full"
              />
              <span
                class="drop-shadow-xl text-center z-10 text-primary-dark text-mobile-page-header font-medium bg-white/75 px-2 rounded-sm w-full py-2"
              >
                Regional info center groups
              </span>
            </div>
          </a>
        </div>
        <div
          class="flex justify-center"
        >
          <a
            href="/"
          >
            <div
              class="drop-shadow-md shrink-0 relative w-[200px] h-[200px] rounded-ml bg-learningSquare opacity-50 cursor-default bg-cover flex flex-col items-center justify-center"
            >
              <div
                class="bg-white/50 absolute rounded-ml w-full h-full"
              />
              <span
                class="drop-shadow-xl text-center z-10 text-primary-dark text-mobile-page-header font-medium bg-white/75 px-2 rounded-sm w-full py-2"
              >
                Japanese town resources
              </span>
            </div>
          </a>
        </div>
        <div
          class="flex justify-center"
        >
          <a
            href="/"
          >
            <div
              class="drop-shadow-md shrink-0 relative w-[200px] h-[200px] rounded-ml bg-learningGlobeSquare opacity-50 cursor-default bg-cover flex flex-col items-center justify-center"
            >
              <div
                class="bg-white/50 absolute rounded-ml w-full h-full"
              />
              <span
                class="drop-shadow-xl text-center z-10 text-primary-dark text-mobile-page-header font-medium bg-white/75 px-2 rounded-sm w-full py-2"
              >
                Japanese town groups
              </span>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
